# Preparation

In [2]:
import warnings
warnings.filterwarnings("ignore")

In [13]:
import pandas as pd
import numpy as np
import datetime as dt

Read data

In [5]:
df = pd.read_csv('../data/NYC_311_subset.csv')
df_pandas = pd.DataFrame(df)
df_pandas

Unique Key            Created Date             Closed Date Agency  \
0          41316662  01/01/2019 12:00:00 AM  01/02/2019 12:00:00 AM  DOHMH   
1          41316968  01/01/2019 12:00:00 AM  01/15/2019 04:28:53 PM  DOHMH   
2          41318587  01/01/2019 12:00:00 AM  01/01/2019 12:00:00 AM  DOHMH   
3          41309122  01/01/2019 12:00:00 AM  01/10/2019 12:00:00 AM  DOHMH   
4          41310910  01/01/2019 12:00:00 AM  01/01/2019 12:00:00 AM  DOHMH   
...             ...                     ...                     ...    ...   
2456120    45288120  01/01/2020 12:00:00 AM  01/02/2020 09:51:29 AM  DOHMH   
2456121    45288240  01/01/2020 12:00:00 AM  01/02/2020 12:00:01 AM  DOHMH   
2456122    45288728  01/01/2020 12:00:00 AM  01/02/2020 12:00:01 AM  DOHMH   
2456123    45289555  01/01/2020 12:00:00 AM  01/02/2020 09:39:33 AM  DOHMH   
2456124    45289558  01/01/2020 12:00:00 AM  01/15/2020 12:00:01 AM  DOHMH   

                                     Agency Name  Complaint Type  \
0        Department of Health and Mental Hygiene          Rodent   
1        Department of Health and Mental Hygiene          Rodent   
2        Department of Health and Mental Hygiene          Rodent   
3        Department of Health and Mental Hygiene          Rodent   
4        Department of Health and Mental Hygiene          Rodent   
...                                          ...             ...   
2456120  Department of Health and Mental Hygiene  Food Poisoning   
2456121  Department of Health and Mental Hygiene  Food Poisoning   
2456122  Department of Health and Mental Hygiene  Food Poisoning   
2456123  Department of Health and Mental Hygiene  Food Poisoning   
2456124  Department of Health and Mental Hygiene  Food Poisoning   

             Descriptor               Location Type Incident Zip  \
0          Rat Sighting          Parking Lot/Garage        11374   
1          Rat Sighting     3+ Family Apt. Building        11219   
2          Rat Sighting     3+ Family Apt. Building        10456   
3        Mouse Sighting     3+ Family Apt. Building        11373   
4          Rat Sighting     3+ Family Apt. Building        10459   
...                 ...                         ...          ...   
2456120       3 or More  Restaurant/Bar/Deli/Bakery        10455   
2456121          1 or 2  Restaurant/Bar/Deli/Bakery        11385   
2456122          1 or 2  Restaurant/Bar/Deli/Bakery        11225   
2456123          1 or 2  Restaurant/Bar/Deli/Bakery        11421   
2456124       3 or More       Other (Explain Below)        11215   

                 Incident Address  ... Vehicle Type Taxi Company Borough  \
0        62-54 JUNCTION BOULEVARD  ...          NaN                  NaN   
1         5906 NEW UTRECHT AVENUE  ...          NaN                  NaN   
2            1220 GRAND CONCOURSE  ...          NaN                  NaN   
3                 90-34 52 AVENUE  ...          NaN                  NaN   
4             899 EAST 169 STREET  ...          NaN                  NaN   
...                           ...  ...          ...                  ...   
2456120       748 EAST 149 STREET  ...          NaN                  NaN   
2456121      1717 CORNELIA STREET  ...          NaN                  NaN   
2456122       985 NOSTRAND AVENUE  ...          NaN                  NaN   
2456123      92-20 JAMAICA AVENUE  ...          NaN                  NaN   
2456124          625 UNION STREET  ...          NaN                  NaN   

        Taxi Pick Up Location Bridge Highway Name Bridge Highway Direction  \
0                         NaN                 NaN                      NaN   
1                         NaN                 NaN                      NaN   
2                         NaN                 NaN                      NaN   
3                         NaN                 NaN                      NaN   
4                         NaN                 NaN                      NaN   
...                       ...                 ...        

check data
+ check data type
+ check length
+ check headers

In [15]:
df_pandas.dtypes

Unique Key                          int64
Created Date                       object
Closed Date                        object
Agency                             object
Agency Name                        object
Complaint Type                     object
Descriptor                         object
Location Type                      object
Incident Zip                       object
Incident Address                   object
Street Name                        object
Cross Street 1                     object
Cross Street 2                     object
Intersection Street 1              object
Intersection Street 2              object
Address Type                       object
City                               object
Landmark                           object
Facility Type                      object
Status                             object
Due Date                           object
Resolution Description             object
Resolution Action Updated Date     object
Community Board                   

In [6]:
print (len(df))

2456125


In [8]:
print ("Column names: ",df.columns.tolist())

Column names:  ['Unique Key', 'Created Date', 'Closed Date', 'Agency', 'Agency Name', 'Complaint Type', 'Descriptor', 'Location Type', 'Incident Zip', 'Incident Address', 'Street Name', 'Cross Street 1', 'Cross Street 2', 'Intersection Street 1', 'Intersection Street 2', 'Address Type', 'City', 'Landmark', 'Facility Type', 'Status', 'Due Date', 'Resolution Description', 'Resolution Action Updated Date', 'Community Board', 'BBL', 'Borough', 'X Coordinate (State Plane)', 'Y Coordinate (State Plane)', 'Open Data Channel Type', 'Park Facility Name', 'Park Borough', 'Vehicle Type', 'Taxi Company Borough', 'Taxi Pick Up Location', 'Bridge Highway Name', 'Bridge Highway Direction', 'Road Ramp', 'Bridge Highway Segment', 'Latitude', 'Longitude', 'Location']


# Overview of NYC311 dataset

clean data for general analysis

In [9]:
Data1 = df_pandas[['Created Date','Closed Date','Agency','Agency Name','Complaint Type','Descriptor','Borough','Incident Zip']]

clean columns and data types

In [10]:
Data1.rename(columns = {'Incident Zip':'Zipcode'}, inplace = True) 
Data1['Zipcode'] = Data1['Zipcode'].astype(str).str[0:5]

In [11]:
Data1['Created Date'] = pd.to_datetime(Data1['Created Date'],format ='%m/%d/%Y %I:%M:%S %p')
Data1['Closed Date'] = pd.to_datetime(Data1['Closed Date'],format ='%m/%d/%Y %I:%M:%S %p')

In [12]:
Data1.dtypes

Created Date      datetime64[ns]
Closed Date       datetime64[ns]
Agency                    object
Agency Name               object
Complaint Type            object
Descriptor                object
Borough                   object
Zipcode                   object
dtype: object

Calculate and check ticket Open days

In [14]:
Data1.insert(2, "Open days", Data1['Closed Date'] - Data1['Created Date'], True)
Data1

Created Date         Closed Date        Open days Agency  \
0         2019-01-01 2019-01-02 00:00:00  1 days 00:00:00  DOHMH   
1         2019-01-01 2019-01-15 16:28:53 14 days 16:28:53  DOHMH   
2         2019-01-01 2019-01-01 00:00:00  0 days 00:00:00  DOHMH   
3         2019-01-01 2019-01-10 00:00:00  9 days 00:00:00  DOHMH   
4         2019-01-01 2019-01-01 00:00:00  0 days 00:00:00  DOHMH   
...              ...                 ...              ...    ...   
2456120   2020-01-01 2020-01-02 09:51:29  1 days 09:51:29  DOHMH   
2456121   2020-01-01 2020-01-02 00:00:01  1 days 00:00:01  DOHMH   
2456122   2020-01-01 2020-01-02 00:00:01  1 days 00:00:01  DOHMH   
2456123   2020-01-01 2020-01-02 09:39:33  1 days 09:39:33  DOHMH   
2456124   2020-01-01 2020-01-15 00:00:01 14 days 00:00:01  DOHMH   

                                     Agency Name  Complaint Type  \
0        Department of Health and Mental Hygiene          Rodent   
1        Department of Health and Mental Hygiene          Rodent   
2        Department of Health and Mental Hygiene          Rodent   
3        Department of Health and Mental Hygiene          Rodent   
4        Department of Health and Mental Hygiene          Rodent   
...                                          ...             ...   
2456120  Department of Health and Mental Hygiene  Food Poisoning   
2456121  Department of Health and Mental Hygiene  Food Poisoning   
2456122  Department of Health and Mental Hygiene  Food Poisoning   
2456123  Department of Health and Mental Hygiene  Food Poisoning   
2456124  Department of Health and Mental Hygiene  Food Poisoning   

             Descriptor   Borough Zipcode  
0          Rat Sighting    QUEENS   11374  
1          Rat Sighting  BROOKLYN   11219  
2          Rat Sighting     BRONX   10456  
3        Mouse Sighting    QUEENS   11373  
4          Rat Sighting     BRONX   10459  
...                 ...       ...     ...  
2456120       3 or More     BRONX   10455  
2456121          1 or 2    QUEENS   11385  
2456122          1 or 2  BROOKLYN   11225  
2456123          1 or 2    QUEENS   11421  
2456124       3 or More  BROOKLYN   11215  

[2456125 rows x 9 columns]

In [156]:
Data1[Data1['Open days']< '0 days'].describe()

Open days
count                         15345
mean   -13 days +09:05:52.666471164
std     395 days 02:05:12.525401268
min           -43492 days +07:04:34
25%               -4 days +00:00:00
50%               -3 days +00:00:00
75%               -3 days +14:58:00
max               -1 days +23:59:23

In [155]:
Data1[Data1['Open days']>= '0 days'].describe()

Open days
count                     2368536
mean   10 days 16:26:25.546276691
std    33 days 04:53:29.032872985
min               0 days 00:00:00
25%               0 days 02:42:35
50%               0 days 21:04:13
75%        5 days 01:24:58.500000
max             610 days 09:13:15

Case numbers by Agency

In [71]:
Data1['Agency Name'].describe()

count                             2456125
unique                                412
top       New York City Police Department
freq                               840353
Name: Agency Name, dtype: object

In [105]:
print (Data1['Agency Name'].unique())

['Department of Health and Mental Hygiene'
 'Department of Housing Preservation and Development'
 'New York City Police Department' 'Taxi and Limousine Commission'
 'Department of Transportation' 'Department of Environmental Protection'
 'Department of Sanitation' 'Department of Buildings'
 'BCC - Brooklyn South' 'Department of Parks and Recreation'
 'HRA Benefit Card Replacement' 'Department of Consumer Affairs'
 'Operations Unit - Department of Homeless Services' 'BCC - Staten Island'
 'BCC - Bronx' 'A - Brooklyn' 'A - Manhattan' 'BCC - Queens East'
 'Manhattan 08' 'Queens West 02' 'Bronx 08'
 'A - Illegal Posting Staten Island, Queens and Brooklyn'
 'BCC - Brooklyn North' 'Bronx 03' 'BCC - Queens West' 'Queens West 03'
 'Manhattan 03' 'Brooklyn South 10' 'Correspondence Unit'
 'Brooklyn North 03' 'Bronx 10' 'A - Canine Task Force Citywide'
 'Queens East 12' 'A - Staten Island' 'Queens East 10' 'BCC - Manhattan'
 'A - Queens' 'Manhattan 01' 'Adjudication - Hearing by Mail'
 'Departme

In [103]:
Data2 = Data1['Agency Name'].value_counts(ascending=False)
Data2

New York City Police Department                       840353
Department of Housing Preservation and Development    465672
Department of Transportation                          281864
Department of Environmental Protection                180227
Department of Sanitation                              141478
                                                       ...  
School - Kappa                                             1
School - PS 38 Roberto Clemente                            1
School - IS 181 Pablo Casals                               1
School - IS 27 Anning S Prall                              1
School - PS 233 Langston Hughes                            1
Name: Agency Name, Length: 412, dtype: int64

In [122]:
Agency_Percentage = pd.DataFrame(Data1['Agency Name'].value_counts(normalize=True)*100,)
Agency_Percentage[Agency_Percentage['Agency Name']>1]

Agency Name
New York City Police Department                       34.214586
Department of Housing Preservation and Development    18.959621
Department of Transportation                          11.475963
Department of Environmental Protection                 7.337859
Department of Sanitation                               5.760212
Department of Buildings                                5.087526
Department of Parks and Recreation                     4.190259
Department of Health and Mental Hygiene                2.623075
Taxi and Limousine Commission                          1.595155

In [123]:
sum(Agency_Percentage['Agency Name'])

99.99999999999888

Case numbers by Borough

In [81]:
Data1['Borough'].describe()

count      2456125
unique           6
top       BROOKLYN
freq        772368
Name: Borough, dtype: object

In [94]:
Borough1 = Data1.groupby('Borough').count()
Borough2 = Borough1.iloc[0:5]
Borough2.sort_values(by='Created Date',ascending = False)

Created Date  Closed Date  Open days  Agency  Agency Name  \
Borough                                                                    
BROOKLYN             772368       750108     750108  772368       772368   
QUEENS               605305       589535     589535  605305       605305   
MANHATTAN            488027       467415     467415  488027       488027   
BRONX                439094       431923     431923  439094       439094   
STATEN ISLAND        129122       125042     125042  129122       129122   

               Complaint Type  Descriptor  Zipcode  
Borough                                             
BROOKLYN               772368      762211   772368  
QUEENS                 605305      600511   605305  
MANHATTAN              488027      465301   488027  
BRONX                  439094      434374   439094  
STATEN ISLAND          129122      128326   129122

In [130]:
Borough_Percentage = Data1['Borough'].value_counts(normalize=True)*100
Borough_Percentage

BROOKLYN         31.446608
QUEENS           24.644715
MANHATTAN        19.869795
BRONX            17.877510
STATEN ISLAND     5.257143
Unspecified       0.904229
Name: Borough, dtype: float64

Mean case numbers for Boroughs

In [54]:
Borough2.mean(axis=0)

Created Date      486783.2
Closed Date       472804.6
Open days         472804.6
Agency            486783.2
Agency Name       486783.2
Complaint Type    486783.2
Descriptor        478144.6
Zipcode           486783.2
dtype: float64

Check boroughs that have over average cases

In [95]:
Borough2[Borough2['Zipcode']>486783.2]

Created Date  Closed Date  Open days  Agency  Agency Name  \
Borough                                                                
BROOKLYN         772368       750108     750108  772368       772368   
MANHATTAN        488027       467415     467415  488027       488027   
QUEENS           605305       589535     589535  605305       605305   

           Complaint Type  Descriptor  Zipcode  
Borough                                         
BROOKLYN           772368      762211   772368  
MANHATTAN          488027      465301   488027  
QUEENS             605305      600511   605305

## Results for General analysis
+ a total of 2456125 case recorded
+ 2383881 cases have open days records with a mean ticket open days around 10 days
+ the mean value for cases by each agency
+ 412 unique Agency has been reported to. The agency that received most reports is New York City Police Department
+ percentage of top several agencies cases
+ Brooklyn have the largest case number: 772368 cases for the past year.
+ Brooklyn, Queens and Manhattan have larger case numbers than the average for the past year.
+ The average number of cases for five boroughs is 486783.2 from January 2019 to January 2020.

# NYC311 analysis for DSNY

filter data for DSNY

In [285]:
DSNY_311 = Data1[Data1['Agency Name']=='Department of Sanitation']
DSNY_311

Created Date Closed Date         Open days Agency  \
68      2019-01-01 00:06:00  2019-01-10   8 days 23:54:00   DSNY   
202     2019-01-01 00:38:00  2019-01-03   1 days 23:22:00   DSNY   
581     2019-01-01 01:41:00  2019-01-03   1 days 22:19:00   DSNY   
1020    2019-01-01 03:19:36  2019-05-08 126 days 20:40:24   DSNY   
1253    2019-01-01 04:40:00  2019-01-03   1 days 19:20:00   DSNY   
...                     ...         ...               ...    ...   
2454322 2019-12-31 15:24:08         NaT               NaT   DSNY   
2454737 2019-12-31 16:46:55         NaT               NaT   DSNY   
2455286 2019-12-31 19:31:33         NaT               NaT   DSNY   
2455691 2019-12-31 22:02:10         NaT               NaT   DSNY   
2455696 2019-12-31 22:04:16         NaT               NaT   DSNY   

                      Agency Name                       Complaint Type  \
68       Department of Sanitation  Request Large Bulky Item Collection   
202      Department of Sanitation  Request Large Bulky Item Collection   
581      Department of Sanitation  Request Large Bulky Item Collection   
1020     Department of Sanitation                             Graffiti   
1253     Department of Sanitation  Request Large Bulky Item Collection   
...                           ...                                  ...   
2454322  Department of Sanitation                             Graffiti   
2454737  Department of Sanitation                             Graffiti   
2455286  Department of Sanitation                             Graffiti   
2455691  Department of Sanitation                             Graffiti   
2455696  Department of Sanitation                             Graffiti   

                                  Descriptor    Borough Zipcode  
68       Request Large Bulky Item Collection   BROOKLYN   11229  
202      Request Large Bulky Item Collection      BRONX   10469  
581      Request Large Bulky Item Collection  MANHATTAN   10017  
1020                                Graffiti   BROOKLYN   11205  
1253     Request Large Bulky Item Collection   BROOKLYN   11233  
...                                      ...        ...     ...  
2454322                             Graffiti   BROOKLYN   11209  
2454737                             Graffiti      BRONX   10457  
2455286                             Graffiti  MANHATTAN   10011  
2455691                             Graffiti  MANHATTAN   10016  
2455696                             Graffiti  MANHATTAN   10016  

[141478 rows x 9 columns]

check distribution and basic stats

In [286]:
DSNY_311['Borough'].describe()

count       141478
unique           6
top       BROOKLYN
freq         55713
Name: Borough, dtype: object

In [287]:
Borough_Percentage1 = DSNY_311['Borough'].value_counts(normalize=True)*100
Borough_Percentage1

BROOKLYN         39.379267
QUEENS           31.307341
MANHATTAN        11.555153
STATEN ISLAND    10.754322
BRONX             6.993313
Unspecified       0.010602
Name: Borough, dtype: float64

In [288]:
DSNY_311[DSNY_311['Open days']<'0 days'].describe()

Open days
count                             21
mean   -278 days +13:36:06.952380952
std      556 days 17:11:43.791406544
min             -2398 days +17:38:46
25%              -143 days +00:00:00
50%              -143 days +00:41:00
75%                -4 days +00:32:00
max                -4 days +00:41:00

In [289]:
DSNY_311[DSNY_311['Open days']=='0 days'].describe()

Open days
count              135
mean   0 days 00:00:00
std    0 days 00:00:00
min    0 days 00:00:00
25%    0 days 00:00:00
50%    0 days 00:00:00
75%    0 days 00:00:00
max    0 days 00:00:00

In [290]:
DSNY_311[DSNY_311['Open days']>='0 days'].describe()

Open days
count                      134376
mean   19 days 03:23:16.220619749
std    41 days 14:15:10.508571206
min               0 days 00:00:00
25%               2 days 00:19:00
50%               3 days 12:22:00
75%               7 days 02:52:00
max             423 days 15:41:27

In [291]:
DSNY_311[DSNY_311['Open days']>='300 days'].sort_values(by='Open days' ,ascending = True)

Created Date         Closed Date         Open days Agency  \
75357  2019-01-11 12:37:44 2019-11-14 00:00:00 306 days 11:22:16   DSNY   
104392 2019-01-15 10:04:56 2019-11-18 00:00:00 306 days 13:55:04   DSNY   
103413 2019-01-15 08:22:18 2019-11-19 00:00:00 307 days 15:37:42   DSNY   
67467  2019-01-10 13:12:20 2019-11-18 00:00:00 311 days 10:47:40   DSNY   
73132  2019-01-11 09:29:10 2019-11-19 12:46:23 312 days 03:17:13   DSNY   
41802  2019-01-07 10:43:22 2019-11-19 12:46:23 316 days 02:03:01   DSNY   
7947   2019-01-02 11:19:25 2019-11-19 12:46:23 321 days 01:26:58   DSNY   
7945   2019-01-02 11:19:18 2019-11-19 12:46:23 321 days 01:27:05   DSNY   
43000  2019-01-07 12:28:05 2020-01-02 00:00:00 359 days 11:31:55   DSNY   
411216 2019-02-25 08:40:01 2020-03-18 00:00:00 386 days 15:19:59   DSNY   
40325  2019-01-07 08:18:33 2020-03-06 00:00:00 423 days 15:41:27   DSNY   

                     Agency Name Complaint Type Descriptor    Borough Zipcode  
75357   Department of Sanitation       Graffiti   Graffiti  MANHATTAN   10009  
104392  Department of Sanitation       Graffiti   Graffiti   BROOKLYN     nan  
103413  Department of Sanitation       Graffiti   Graffiti   BROOKLYN   11232  
67467   Department of Sanitation       Graffiti   Graffiti   BROOKLYN   11226  
73132   Department of Sanitation       Graffiti   Graffiti  MANHATTAN   10002  
41802   Department of Sanitation       Graffiti   Graffiti      BRONX   10452  
7947    Department of Sanitation       Graffiti   Graffiti     QUEENS   11372  
7945    Department of Sanitation       Graffiti   Graffiti     QUEENS   11372  
43000   Department of Sanitation       Graffiti   Graffiti   BROOKLYN   11211  
411216  Department of Sanitation       Graffiti   Graffiti   BROOKLYN   11234  
40325   Department of Sanitation       Graffiti   Graffiti   BROOKLYN   11206

Check stats for Complaint Types

In [292]:
print (DSNY_311['Complaint Type'].value_counts())

Request Large Bulky Item Collection    99538
Graffiti                               22289
Electronics Waste Appointment          18808
Missed Collection                        340
Missed Collection (All Materials)        165
Sanitation Condition                     105
Snow Removal                              59
Other Enforcement                         55
Dirty Conditions                          32
Sweeping/Missed-Inadequate                30
Litter Basket / Request                   15
Vacant Lot                                15
Collection Truck Noise                    15
Derelict Vehicles                         12
Name: Complaint Type, dtype: int64


In [293]:
Type_Percentage = DSNY_311['Complaint Type'].value_counts(normalize=True)*100
Type_Percentage = pd.DataFrame(Type_Percentage)
Type_Percentage

Complaint Type
Request Large Bulky Item Collection       70.355815
Graffiti                                  15.754393
Electronics Waste Appointment             13.293940
Missed Collection                          0.240320
Missed Collection (All Materials)          0.116626
Sanitation Condition                       0.074216
Snow Removal                               0.041703
Other Enforcement                          0.038875
Dirty Conditions                           0.022618
Sweeping/Missed-Inadequate                 0.021205
Litter Basket / Request                    0.010602
Vacant Lot                                 0.010602
Collection Truck Noise                     0.010602
Derelict Vehicles                          0.008482

In [294]:
Top = DSNY_311[DSNY_311['Complaint Type']=='Request Large Bulky Item Collection']
Top['Open days'].describe()

count                        99538
mean     3 days 17:59:04.687255058
std      3 days 02:51:58.859006729
min                0 days 00:01:00
25%                1 days 13:57:00
50%                3 days 01:05:00
75%                4 days 14:16:45
max              151 days 08:26:00
Name: Open days, dtype: object

In [297]:
Top2 = DSNY_311[DSNY_311['Complaint Type']=='Graffiti']
Top2[Top2['Open days']>='0 days'].describe()

Open days
count                        15741
mean   125 days 23:38:55.893717066
std     39 days 05:44:36.930283314
min                0 days 00:00:00
25%               96 days 14:05:22
50%              124 days 08:08:08
75%              151 days 16:44:31
max              423 days 15:41:27

In [299]:
Top3 = DSNY_311[DSNY_311['Complaint Type']=='Electronics Waste Appointment']
Top3['Open days'].describe()

count                         18808
mean     11 days 08:55:49.679923437
std      12 days 02:44:56.325524596
min                 0 days 00:00:00
25%                 2 days 02:25:45
50%                 5 days 04:31:00
75%                20 days 22:28:45
max               141 days 19:50:00
Name: Open days, dtype: object

In [296]:
Last = DSNY_311[DSNY_311['Complaint Type']=='Derelict Vehicles']
Last['Open days'].describe()

count                           12
mean               0 days 01:48:35
std      0 days 01:50:50.999241399
min                0 days 00:00:00
25%                0 days 00:00:00
50%                0 days 01:44:30
75%                0 days 03:09:30
max                0 days 05:09:00
Name: Open days, dtype: object

# Load other data

In [192]:
Rbin = pd.read_csv('../data/Public_Recycling_Bins.csv')

In [213]:
Mton = pd.read_csv('../data/DSNY_Monthly_Tonnage_Data.csv')
Mton.dtypes

MONTH                   object
BOROUGH                 object
COMMUNITYDISTRICT        int64
REFUSETONSCOLLECTED    float64
PAPERTONSCOLLECTED     float64
MGPTONSCOLLECTED       float64
RESORGANICSTONS        float64
SCHOOLORGANICTONS      float64
LEAVESORGANICTONS      float64
XMASTREETONS           float64
BOROUGH_ID               int64
dtype: object

In [214]:
Mton

MONTH    BOROUGH  COMMUNITYDISTRICT  REFUSETONSCOLLECTED  \
0      1993 / 11  Manhattan                  1                625.2   
1      1992 / 03  Manhattan                  1                726.0   
2      1995 / 08  Manhattan                  1                627.5   
3      1991 / 08  Manhattan                  1                695.1   
4      1995 / 09  Manhattan                  1                633.1   
...          ...        ...                ...                  ...   
21043  2017 / 07      Bronx                  6               2143.5   
21044  2019 / 02      Bronx                  6               1921.2   
21045  2018 / 10      Bronx                  6               2281.1   
21046  2017 / 12      Bronx                  6               2087.1   
21047  2019 / 04      Bronx                  6               2246.0   

       PAPERTONSCOLLECTED  MGPTONSCOLLECTED  RESORGANICSTONS  \
0                   119.4              34.4              NaN   
1                     NaN               NaN              NaN   
2                    82.4              35.6              NaN   
3                     NaN               NaN              NaN   
4                   117.2              45.8              NaN   
...                   ...               ...              ...   
21043               118.8             177.1              NaN   
21044               117.2             167.9             10.6   
21045               145.2             215.7              8.1   
21046               132.1             165.0              NaN   
21047               144.8             195.8             12.1   

       SCHOOLORGANICTONS  LEAVESORGANICTONS  XMASTREETONS  BOROUGH_ID  
0                    NaN                NaN           NaN           1  
1                    NaN                NaN           NaN           1  
2                    NaN                NaN           NaN           1  
3                    NaN                NaN           NaN           1  
4                    NaN                NaN           NaN           1  
...                  ...                ...           ...         ...  
21043                NaN                NaN           NaN           2  
21044               29.7                NaN           NaN           2  
21045               73.9                NaN           NaN           2  
21046               41.2                NaN           NaN           2  
21047               34.7                NaN           NaN           2  

[21048 rows x 11 columns]

check recycling bin and overflow

In [185]:
Rbin['Borough'].value_counts()

Manhattan        184
Queens           117
Bronx            108
Brooklyn          89
Staten Island     42
Brooklyn           5
Name: Borough, dtype: int64

In [191]:
Overflow = DSNY_311[DSNY_311['Complaint Type']=='Overflowing Recycling Baskets']
Overflow['Borough'].value_counts()

BROOKLYN         2
STATEN ISLAND    1
Name: Borough, dtype: int64

Monthly situation check

In [225]:
Mton['MONTH'] = pd.to_datetime(Mton['MONTH'],format ='%Y / %m')
Mton_2019=Mton[Mton['MONTH'].dt.year==2019]
Mton_2019

MONTH        BOROUGH  COMMUNITYDISTRICT  REFUSETONSCOLLECTED  \
55    2019-10-01       Brooklyn                  7               3342.5   
56    2019-08-01       Brooklyn                  7               3263.2   
87    2019-02-01  Staten Island                  3               4392.4   
91    2019-08-01  Staten Island                  3               5757.2   
92    2019-04-01  Staten Island                  3               6037.6   
...          ...            ...                ...                  ...   
20904 2019-07-01      Manhattan                  1               1366.7   
20948 2019-04-01       Brooklyn                  4               3038.9   
20994 2019-05-01          Bronx                  3               1985.8   
21044 2019-02-01          Bronx                  6               1921.2   
21047 2019-04-01          Bronx                  6               2246.0   

       PAPERTONSCOLLECTED  MGPTONSCOLLECTED  RESORGANICSTONS  \
55                  418.4             415.4             79.9   
56                  399.4             408.1             69.7   
87                  638.3             586.1              NaN   
91                  731.6             756.4              NaN   
92                  716.8             716.2              NaN   
...                   ...               ...              ...   
20904               362.0             220.7             26.4   
20948               252.6             349.1              NaN   
20994               127.7             195.9              NaN   
21044               117.2             167.9             10.6   
21047               144.8             195.8             12.1   

       SCHOOLORGANICTONS  LEAVESORGANICTONS  XMASTREETONS  BOROUGH_ID  
55                 118.9                NaN           NaN           3  
56                   NaN                NaN           NaN           3  
87                  19.2                NaN           1.0           5  
91                   NaN                NaN           NaN           5  
92                  21.4                NaN           NaN           5  
...                  ...                ...           ...         ...  
20904                NaN                NaN           NaN           1  
20948                NaN                NaN           NaN           3  
20994                NaN                NaN           NaN           2  
21044               29.7                NaN           NaN           2  
21047               34.7                NaN           NaN           2  

[708 rows x 11 columns]

In [232]:
Tonnage_Borough = Mton_2019[['MONTH','BOROUGH']]

In [226]:
Tonnage = Mton_2019[['REFUSETONSCOLLECTED','PAPERTONSCOLLECTED','MGPTONSCOLLECTED','RESORGANICSTONS','SCHOOLORGANICTONS','LEAVESORGANICTONS','XMASTREETONS']]
Tonnage

REFUSETONSCOLLECTED  PAPERTONSCOLLECTED  MGPTONSCOLLECTED  \
55                  3342.5               418.4             415.4   
56                  3263.2               399.4             408.1   
87                  4392.4               638.3             586.1   
91                  5757.2               731.6             756.4   
92                  6037.6               716.8             716.2   
...                    ...                 ...               ...   
20904               1366.7               362.0             220.7   
20948               3038.9               252.6             349.1   
20994               1985.8               127.7             195.9   
21044               1921.2               117.2             167.9   
21047               2246.0               144.8             195.8   

       RESORGANICSTONS  SCHOOLORGANICTONS  LEAVESORGANICTONS  XMASTREETONS  
55                79.9              118.9                NaN           NaN  
56                69.7                NaN                NaN           NaN  
87                 NaN               19.2                NaN           1.0  
91                 NaN                NaN                NaN           NaN  
92                 NaN               21.4                NaN           NaN  
...                ...                ...                ...           ...  
20904             26.4                NaN                NaN           NaN  
20948              NaN                NaN                NaN           NaN  
20994              NaN                NaN                NaN           NaN  
21044             10.6               29.7                NaN           NaN  
21047             12.1               34.7                NaN           NaN  

[708 rows x 7 columns]

In [227]:
Tonnage_sum = Tonnage.sum(axis=1)
Tonnage_sum

55       4375.1
56       4140.4
87       5637.0
91       7245.2
92       7492.0
          ...  
20904    1975.8
20948    3640.6
20994    2309.4
21044    2246.6
21047    2633.4
Length: 708, dtype: float64

In [249]:
Borough_tonnage_sum = pd.concat([Tonnage_Borough, Tonnage_sum], axis=1)
Borough_tonnage_sum

MONTH        BOROUGH       0
55    2019-10-01       Brooklyn  4375.1
56    2019-08-01       Brooklyn  4140.4
87    2019-02-01  Staten Island  5637.0
91    2019-08-01  Staten Island  7245.2
92    2019-04-01  Staten Island  7492.0
...          ...            ...     ...
20904 2019-07-01      Manhattan  1975.8
20948 2019-04-01       Brooklyn  3640.6
20994 2019-05-01          Bronx  2309.4
21044 2019-02-01          Bronx  2246.6
21047 2019-04-01          Bronx  2633.4

[708 rows x 3 columns]

In [253]:
Manhattan_tonnage = Borough_tonnage_sum[(Borough_tonnage_sum['BOROUGH']=='Manhattan')].groupby('MONTH').sum('0')

In [252]:
Brooklyn_tonnage = Borough_tonnage_sum[(Borough_tonnage_sum['BOROUGH']=='Brooklyn')].groupby('MONTH').sum('0')

In [254]:
Queens_tonnage = Borough_tonnage_sum[(Borough_tonnage_sum['BOROUGH']=='Queens')].groupby('MONTH').sum('0')

In [256]:
StatenIsland_tonnage = Borough_tonnage_sum[(Borough_tonnage_sum['BOROUGH']=='Staten Island')].groupby('MONTH').sum('0')

In [257]:
Bronx_tonnage = Borough_tonnage_sum[(Borough_tonnage_sum['BOROUGH']=='Bronx')].groupby('MONTH').sum('0')

In [264]:
Mton_cleaned = pd.concat([Manhattan_tonnage, Brooklyn_tonnage,Queens_tonnage,StatenIsland_tonnage,Bronx_tonnage], axis=1)
Mton_cleaned.columns=['Manhattan','Brooklyn','Queens','Staten Island','Bronx']
Mton_cleaned['Total'] = Mton_cleaned.sum(axis=1)
Mton_cleaned.sort_values(by='Total',ascending = True)

Manhattan  Brooklyn   Queens  Staten Island    Bronx     Total
MONTH                                                                     
2019-02-01    41047.4   70650.2  59192.9        14987.7  35191.3  221069.5
2019-03-01    44238.9   78293.3  65447.9        16799.7  39344.3  244124.1
2019-11-01    45738.4   81120.5  71678.3        18117.2  39006.8  255661.2
2019-09-01    44433.9   82128.7  71885.1        18574.2  39493.5  256515.4
2019-01-01    47816.0   83235.1  70655.3        18043.8  40867.2  260617.4
2019-04-01    45979.6   84233.5  73752.1        19832.8  41556.3  265354.3
2019-08-01    45339.1   82368.5  77339.1        19732.8  42731.8  267511.3
2019-12-01    47045.9   87404.8  76752.1        20585.5  41406.2  273194.5
2019-10-01    48543.7   87475.5  76552.9        19386.9  42021.4  273980.4
2019-06-01    46483.2   87301.0  78697.4        20990.7  41878.3  275350.6
2019-07-01    45521.0   86447.7  81653.6        21196.5  44572.4  279391.2
2019-05-01    49343.2   89066.4  80408.7        22127.2  43350.9  284296.4

In [273]:
DSNY_bymonth=DSNY_311.groupby(DSNY_311['Created Date'].dt.strftime('%m')).count()
DSNY_bymonth=DSNY_bymonth[['Created Date']]
DSNY_bymonth.columns=['Case Number']
DSNY_bymonth.sort_values(by='Case Number',ascending = True)

Case Number
Created Date             
11                   9724
10                   9961
09                  10282
08                  11300
12                  11787
07                  12382
02                  29244
03                  35374
01                  35566
04                  37166
05                  40185
06                  40539